integrations-and-supported-tools/detectron2/notebooks/Neptune_detectron2.ipynb

```diff
@@ -0,0 +1,490 @@
+    "\n",
+    "from detectron2.structures import BoxMode\n",
+    "\n",
+    "\n",
+    "def get_balloon_dicts(img_dir):\n",
+    "    json_file = os.path.join(img_dir, \"via_region_data.json\")\n",
+    "    with open(json_file) as f:\n",
+    "        imgs_anns = json.load(f)\n",
+    "\n",
+    "    dataset_dicts = []\n",
+    "    for idx, v in enumerate(imgs_anns.values()):\n",
+    "        record = {}\n",
+    "\n",
+    "        filename = os.path.join(img_dir, v[\"filename\"])\n",
+    "        height, width = cv2.imread(filename).shape[:2]\n",
+    "\n",
+    "        record[\"file_name\"] = filename\n",
+    "        record[\"image_id\"] = idx\n",
+    "        record[\"height\"] = height\n",
+    "        record[\"width\"] = width\n",
+    "\n",
+    "        annos = v[\"regions\"]\n",
+    "        objs = []\n",
+    "        for _, anno in annos.items():\n",
+    "            assert not anno[\"region_attributes\"]\n",
+    "            anno = anno[\"shape_attributes\"]\n",
+    "            px = anno[\"all_points_x\"]\n",
+    "            py = anno[\"all_points_y\"]\n",
+    "            poly = [(x + 0.5, y + 0.5) for x, y in zip(px, py)]\n",
+    "            poly = [p for x in poly for p in x]\n",
+    "\n",
+    "            obj = {\n",
+    "                \"bbox\": [np.min(px), np.min(py), np.max(px), np.max(py)],\n",
+    "                \"bbox_mode\": BoxMode.XYXY_ABS,\n",
+    "                \"segmentation\": [poly],\n",
+    "                \"category_id\": 0,\n",
+    "            }\n",
+    "            objs.append(obj)\n",
+    "        record[\"annotations\"] = objs\n",
+    "        dataset_dicts.append(record)\n",
+    "    return dataset_dicts\n",
+    "\n",
+    "\n",
+    "for d in [\"train\", \"val\"]:\n",
+    "    DatasetCatalog.register(\"balloon_\" + d, lambda d=d: get_balloon_dicts(\"balloon/\" + d))\n",
+    "    MetadataCatalog.get(\"balloon_\" + d).set(thing_classes=[\"balloon\"])\n",
+    "balloon_metadata = MetadataCatalog.get(\"balloon_train\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure the model for training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from detectron2.engine import DefaultTrainer\n",
+    "\n",
+    "device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
+    "\n",
+    "cfg = get_cfg()\n",
+    "cfg.MODEL.DEVICE = device\n",
+    "cfg.DATALOADER.NUM_WORKERS = 0\n",
+    "cfg.merge_from_file(\n",
+    "    model_zoo.get_config_file(\"COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml\")\n",
+    ")\n",
+    "cfg.DATASETS.TRAIN = (\"balloon_train\",)\n",
+    "cfg.DATASETS.TEST = ()\n",
+    "cfg.DATALOADER.NUM_WORKERS = 2\n",
+    "cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(\n",
+    "    \"COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml\"\n",
+    ")  # Let training initialize from model zoo\n",
```

Comment:
There's quite a lot of stuff in the inline comments below as well. Would it be possible to utilize some text cells, for nicer UX? :)